packages/react/src/runtime/style-cache.tsx

```diff
@@ -7,18 +7,31 @@ import { ProviderComponent, UseCacheHook } from './types';
  * Cache to hold already used styles.
  * React Context on the server - singleton object on the client.
  */
-const Cache: any = isNodeEnvironment() ? createContext<Record<string, true> | null>(null) : {};
+let ClientCache = {};
+const Cache: any = createContext<Record<string, true> | null>(null) || ClientCache;
```

Comment:
this will always use the `createContext` flow - can we revert to what it was previously? it's behind a node environment check so the `create context` flow never makes its way to the client

```suggestion
const Cache: any = isNodeEnvironment() ? createContext<Record<string, true> | null>(null) : {};
```